rct229/rules/section5.py

```diff
@@ -97,3 +104,213 @@ def rule_check(self, context, calc_vals, data=None):
 
 
 # ------------------------
+
+
+class Section5Rule8(RuleDefinitionListIndexedBase):
+    """Rule 8 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelop)"""
+
+    def __init__(self):
+        super(Section5Rule8, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section5Rule8.BuildingRule(),
+            index_rmr="baseline",
+            id="5-8",
+            description="Baseline below-grade walls shall match the appropriate assembly maximum C-factors in Table G3.4-1 through G3.4-8.",
+        )
+
+    class BuildingRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section5Rule8.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                rmr_context="",
+            )
+
+        def get_calc_vals(self, context, data=None):
+            # Climate zone should have only one in RMR
+            climate_zone = find_exactly_one("$..climate_zone", context.baseline)
+            # QNS will there be multiple building scenario?
+            building = find_one("$..buildings[*]", context.baseline)
```

Comment:
setting list_path = "buildings[*]" will set the scope of the context.baseline to buildings.

We need building here because the get_surface_conditioning_category_dict request for building.

Reply:
I was not precise enough in my description. `list_path` must resolve to a list (or lists if more than one RMR is used).  RuleDefinitionListIndexedBase uses these lists to generate a list of UserBaselineProposedVals contexts. It then passes each of those contexts in turn to the nested rule (such as BuildingRule).  Within BuildingRule context.baseline will be a single building as desired.

Reply:
Ah, got it!